File: web-front-reactive/src/main/java/template/modules/admin/command/users/UpdateUserCommand.java
package template.modules.admin.command.users;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;
import template.modules.admin.domain.users.User;
import template.modules.admin.domain.users.UserRepository;

@Service
@AllArgsConstructor
public class UpdateUserCommand {
    private UserRepository userRepository;

    public Mono<User> execute(User user) {
        Mono<User> existingUser = findById(user.getId());
        if (existingUser != null) {
            userRepository.save(user);
            return Mono.just(user);
        } else {
            return Mono.empty();
        }
    }

    private Mono<User> findById(Long id) {
        return userRepository.findById(id);

    }

}

File: web-front-reactive2/src/main/kotlin/template/modules/admin/command/users/UpdateUserCommand.kt
package template.modules.admin.command.users

import kotlinx.coroutines.reactor.awaitSingleOrNull
import org.springframework.stereotype.Service
import template.modules.admin.domain.users.User
import template.modules.admin.domain.users.UserRepository

@Service
class UpdateUserCommand(private val repo: UserRepository) {
    suspend fun execute(id: Long, user: User): User? {
        val existingUser = findById(id)
        return if (existingUser != null) repo.save(user).awaitSingleOrNull() else null
    }

    private suspend fun findById(id: Long) =
        repo.findById(id).awaitSingleOrNull()
}